docs/design/Chem Integration.md

```diff
@@ -0,0 +1,69 @@
+# Loose binary identification with Upgrade Assistant
+
+It is not uncommon in large code bases to have a collection of dependencies that have been persisted as only the `.dll` files with no known provenance. These files are considered "loose binaries". This makes it challenging to identify what a binary is while upgrading where an updated version is often needed to support .NET Standard/Core/5. In order to solve this, this document proposes integration with a data source that can help match these loose assemblies to known NuGet packages.
+
+Chem is a tool developed internally to gain insights from the ecosystem of code attached to .NET. As a result of the data that it gathers, it is able to match loose assemblies with their probable origins.  As customers are looking to move to .NET 5, this may pose a challenge as these versions are highly likely to be .NET Framework compilations that may not work as expected on newer platforms. Often, these libraries have NuGet packages now and newer versions that could work on newer platforms.
```

Comment:
Do we care about the phrase "chem"? I think the doc could be better described as "Loose Assembly Resolution Process"

Reply:
I totally agree with this.

Reply:
+1 I had this lingering thought about not calling it Chem instead call it by a name that specifies what its trying to do at least from a UA perspective.

Reply:
I've removed all references to chem